mesonbuild/mconf.py

```diff
@@ -38,11 +37,28 @@ class ConfException(mesonlib.MesonException):
 
 class Conf:
     def __init__(self, build_dir):
-        self.build_dir = build_dir
-        if not os.path.isdir(os.path.join(build_dir, 'meson-private')):
-            raise ConfException('Directory %s does not seem to be a Meson build directory.' % build_dir)
-        self.build = build.load(self.build_dir)
-        self.coredata = coredata.load(self.build_dir)
+        self.build_dir = os.path.abspath(os.path.realpath(build_dir))
+        if self.build_dir.endswith('/meson.build') or self.build_dir.endswith('\\meson.build') or self.build_dir == 'meson.build':
```

Comment:
This could be made more reliable by doing:
```python
if os.path.basename(self.build_dir) == 'meson.build' or self.build_dir == 'meson.build'
```
or even:
```python
if 'meson.build' in [os.path.basename(self.build_dir), self.build_dir]
```
right?

Reply:
Yes, this makes more sense. I have also fixed it where I initially copy pasted the if from.